codes/ruby/chapter_sorting/bucket_sort.rb

```diff
@@ -0,0 +1,45 @@
+=begin
+File: bucket_sort.rb
+Created Time: 2024-04-17
+Author: Martin Xu (martin.xus@gmail.com)
+=end
+
+### 桶排序 ###
+def bucket_sort(nums)
+  # 初始化 k = n/2 个桶，预期向每个桶分配 2 个元素
+  k = nums.length / 2
+  buckets = []
+  0.upto(k - 1) do |i|
+    buckets << []
+  end
```

Comment:
The `Number#upto(&block)` is unusual in general programming languages. It is better to stick with the consistency of the whole codebase, which uses `for...in range` or `range.each` commonly. 

Btw, here we are initializing an array so we should use `Array` instead.

```suggestion
  buckets = Array.new(k) { [] }
```